packages/core/core/src/AssetGraph.js

```diff
@@ -185,20 +207,14 @@ export default class AssetGraph extends Graph {
 
   invalidateFile(filePath: FilePath) {
     let node = this.getNode(filePath);
-    if (!node) {
+    if (!node || node.type !== 'file') {
       return;
     }
 
-    if (node.type === 'file') {
-      this.invalidateNode(node);
-    }
-
-    if (node.type === 'connected_file') {
-      // Invalidate all file nodes connected to this node.
-      for (let connectedNode of this.getNodesConnectedTo(node)) {
-        if (connectedNode.type === 'file') {
-          this.invalidateNode(connectedNode);
-        }
+    // Invalidate all file nodes connected to this node.
+    for (let connectedNode of this.getNodesConnectedTo(node)) {
```

Comment:
Do we still need the File nodes? ie. just use Asset nodes for the invalidation?

Reply:
Well we use file nodes for connected files (like .babelrc) also, and it's useful to be able to easily find the transformer requests for a file when it changes without searching the entire graph.